core/chains/evm/types/models.go

```diff
@@ -64,16 +68,30 @@ func (h *Head) BlockHash() common.Hash {
 }
 
 func (h *Head) GetParent() commontypes.Head[common.Hash] {
```

Comment:
shouldn't `GetParent()` return `*Head`?

that would avoid all the casting that you need to do later - which we want to avoid

one of the big benefits of generics/constraints is to avoid casting - generics will fail to compile while casting has the potential to panic during runtime

Reply:
I think the interface needs to change to be able to do this. It's the self-referential problem, but seems avoidable with another layer of wrapper :shrug: